controllers/argocd/rolebinding.go

```diff
@@ -312,13 +311,51 @@ func (r *ReconcileArgoCD) reconcileRoleBinding(name string, rules []v1.PolicyRul
 	return nil
 }
 
-func getCustomRoleName(name string) string {
-	if name == common.ArgoCDApplicationControllerComponent {
-		return os.Getenv(common.ArgoCDControllerClusterRoleEnvName)
+// getCustomRoleName returns value of env variables defined for custom cluster role for namespace scoped Argo CD
+func getCustomRoleName(componentName string, cr *argoproj.ArgoCD) string {
+
+	// return env variable value for application-controller
+	if componentName == common.ArgoCDApplicationControllerComponent && cr.Spec.Controller.Env != nil {
+		for _, env := range cr.Spec.Controller.Env {
+			if env.Name == common.ArgoCDControllerClusterRoleEnvName {
+				return env.Value
+			}
+		}
```

Comment:
What if we prioritize `cr.Spec.Controller.Env` over `os.Getenv(common.ArgoCDControllerClusterRoleEnvName)` instead of completely removing it? This way, we can ensure that existing users are not affected by the change.  WDYT?

Reply:
Ahh, it slipped from my mind, I will fix it.

Reply:
It is done, PTAL.